lightning-rapid-gossip-sync/src/processing.rs

```diff
@@ -623,4 +624,57 @@ mod tests {
 			panic!("Unexpected update result: {:?}", update_result)
 		}
 	}
+
+	fn generate_blank_rgs_blob(chain_hash: &BlockHash, timestamp: u32) -> Vec<u8> {
+		let mut blob = GOSSIP_PREFIX.to_vec();
+
+		chain_hash.write(&mut blob).unwrap();
+		timestamp.write(&mut blob).unwrap();
+
+		0u32.write(&mut blob); // node count
+		0u32.write(&mut blob); // announcement count
+		0u32.write(&mut blob); // update count
+
+		blob
+	}
+
+	#[test]
+	fn network_graph_is_not_modified_by_blank_update() {
+		let logger = TestLogger::new();
+		let network_graph = NetworkGraph::new(Network::Bitcoin, &logger);
+
+		assert_eq!(network_graph.read_only().channels().len(), 0);
+
+		// apply initialization blob
+		let rapid_sync = RapidGossipSync::new(&network_graph, &logger);
+		let update_result = rapid_sync.update_network_graph(&VALID_RGS_BINARY);
+		if update_result.is_err() {
+			panic!("Unexpected update result: {:?}", update_result)
+		}
+
+		// apply delta blob
+		let genesis_block = bitcoin::blockdata::constants::genesis_block(Network::Bitcoin).block_hash();
```

Comment:
nit: could move this into `generate_blank_rgs_blob`? Unlikely to have a test case that wants to set its own block hash.